src/order/locally_finite.lean

```diff
@@ -559,10 +558,29 @@ instance : locally_finite_order (with_top α) :=
                                         coe_lt_coe]
     end }
 
-end order_top
+variables (a b : α)
 
-section order_bot
-open with_bot
+@[simp] lemma Icc_coe_top :
+  Icc (a : with_top α) ⊤ = cons ⊤ ((Ici a).map embedding.coe_with_top)
+    (λ h, let ⟨x, _, hx⟩ := mem_map.1 h in coe_ne_top hx) := rfl
+
+@[simp] lemma Icc_coe_coe : Icc (a : with_top α) b = (Icc a b).map embedding.coe_with_top := rfl
+
+@[simp] lemma Ico_coe_top : Ico (a : with_top α) ⊤ = (Ici a).map embedding.coe_with_top := rfl
+@[simp] lemma Ico_coe_coe : Ico (a : with_top α) b = (Ico a b).map embedding.coe_with_top := rfl
+
+@[simp] lemma Ioc_coe_top :
+  Ioc (a : with_top α) ⊤ = cons ⊤ ((Ioi a).map embedding.coe_with_top)
+    (λ h, let ⟨x, _, hx⟩ := mem_map.1 h in coe_ne_top hx) := rfl
+
+@[simp] lemma Ioc_coe_coe : Ioc (a : with_top α) b = (Ioc a b).map embedding.coe_with_top := rfl
+
+@[simp] lemma Ioo_coe_top : Ioo (a : with_top α) ⊤ = (Ioi a).map embedding.coe_with_top := rfl
+@[simp] lemma Ioo_coe_coe : Ioo (a : with_top α) b = (Ioo a b).map embedding.coe_with_top := rfl
```

Comment:
Do we actually want these as `simp` lemmas? if so, they'll obstruct the simp lemmas for `locally_finite_order` from firing.

Reply:
Not sure. @jcommelin, you were the one asking for them, what do you think?

Reply:
I would keep the lemmas, but remove the `@[simp]`